### ISSUE TITLE
pipenv clean picks up packages with mismatched casing between name and directory

### ISSUE THREAD

**Issue by bryanforbes** [pypa/pipenv#2726]
##### Issue description

After installing packages with mismatched casing between the package name and the directory (such as `SQLAlchemy`, `pipenv clean` will remove them.

##### Expected result

`pipenv clean` removes only those packages that are not in `Pipfile.lock`.

##### Actual result

```
> pipenv clean -v
Ignoring 'pip'.
Ignoring 'setuptools'.
Ignoring 'wheel'.
Uninstalling 'SQLAlchemy'…
```

##### Steps to replicate

1. `mkdir test-pipenv`
2. `cd test-pipenv`
3. `pipenv --python 3`
```
Creating a virtualenv for this project…
Pipfile: /Users/bryan/Projects/test-pipenv/Pipfile
Using /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/bin/python3.7 (3.7.0) to create virtualenv…
⠇Running virtualenv with interpreter /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/bin/python3.7
Using base prefix '/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7'
/usr/local/lib/python3.7/site-packages/virtualenv.py:1041: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
New python executable in /Users/bryan/.local/share/virtualenvs/test-pipenv-pMYUstlA/bin/python3.7
Also creating executable in /Users/bryan/.local/share/virtualenvs/test-pipenv-pMYUstlA/bin/python
Installing setuptools, pip, wheel...done.

Virtualenv location: /Users/bryan/.local/share/virtualenvs/test-pipenv-pMYUstlA
Creating a Pipfile for this project…
```
4. `pip install SQLAlchemy`
```
Installing SQLAlchemy…
Collecting SQLAlchemy
Installing collected packages: SQLAlchemy
Successfully installed SQLAlchemy-1.2.10

Adding SQLAlchemy to Pipfile's [packages]…
Pipfile.lock not found, creating…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
Updated Pipfile.lock (614716)!
Installing dependencies from Pipfile.lock (614716)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 1/1 — 00:00:00
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
```
5. `pipenv clean --dry-run`
```
SQLAlchemy
```

-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2018.7.1.dev0'`

Pipenv location: `'/usr/local/lib/python3.7/site-packages/pipenv'`

Python location: `'/usr/local/opt/python/bin/python3.7'`

Python installations found:

  - `3.7.0`: `/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/bin/python3.7`
  - `3.7.0`: `/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/bin/python3.7m`
  - `2.7.15`: `/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.10`: `/usr/bin/pythonw`
  - `2.7.10`: `/System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.7.0',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '17.6.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT '
                     '2018; root:xnu-4570.61.1~1/RELEASE_X86_64',
 'python_full_version': '3.7.0',
 'python_version': '3.7',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `GREP_COLOR`
  - `MANPATH`
  - `LESS_TERMCAP_mb`
  - `TERM_PROGRAM`
  - `LESS_TERMCAP_md`
  - `NVM_CD_FLAGS`
  - `LESS_TERMCAP_me`
  - `TERM`
  - `SHELL`
  - `TMPDIR`
  - `Apple_PubSub_Socket_Render`
  - `TERM_PROGRAM_VERSION`
  - `TERM_SESSION_ID`
  - `LESS_TERMCAP_ue`
  - `USER`
  - `NVM_DIR`
  - `LS_COLORS`
  - `GREP_COLORS`
  - `SSH_AUTH_SOCK`
  - `__CF_USER_TEXT_ENCODING`
  - `PAGER`
  - `TMUX`
  - `LESS_TERMCAP_us`
  - `TERMINFO_DIRS`
  - `PATH`
  - `PWD`
  - `JENV_LOADED`
  - `HOMEBREW_GITHUB_API_TOKEN`
  - `EDITOR`
  - `LANG`
  - `ITERM_PROFILE`
  - `XPC_FLAGS`
  - `TMUX_PANE`
  - `XPC_SERVICE_NAME`
  - `SHLVL`
  - `HOME`
  - `COLORFGBG`
  - `ITERM_SESSION_ID`
  - `LOGNAME`
  - `LESS`
  - `VISUAL`
  - `LESS_TERMCAP_so`
  - `TMUX_PLUGIN_MANAGER_PATH`
  - `NVM_BIN`
  - `BROWSER`
  - `DISPLAY`
  - `SECURITYSESSIONID`
  - `P4CONFIG`
  - `LESS_TERMCAP_se`
  - `COLORTERM`
  - `OLDPWD`
  - `PROMPT_EOL_MARK`
  - `VIRTUAL_ENV_DISABLE_PROMPT`
  - `_`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_PYTHON_PATH`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/Users/bryan/.jenv/shims:/Users/bryan/.jenv/bin:/Users/bryan/bin:/Users/bryan/.dotfiles/bin:/Users/bryan/Projects/pebble/PebbleSDK-current/bin:/usr/local/opt/coreutils/libexec/gnubin:/usr/local/opt/gnu-sed/libexec/gnubin:/Users/bryan/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/usr/local/share/dotnet:/opt/X11/bin:/Users/bryan/.jenv/shims:/Users/bryan/.nvm/versions/node/v8.11.1/bin`
  - `SHELL`: `/usr/local/bin/zsh`
  - `EDITOR`: `vi`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/Users/bryan/Projects/test-pipenv`

---------------------------

Contents of `Pipfile` ('/Users/bryan/Projects/test-pipenv/Pipfile'):

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
sqlalchemy = "*"

[dev-packages]

[requires]
python_version = "3.7"

```

Contents of `Pipfile.lock` ('/Users/bryan/Projects/test-pipenv/Pipfile.lock'):

```json
{
    "_meta": {
        "hash": {
            "sha256": "0a19757173776bcff4b84a39482ac8017abed7b06ec58418e17f35c407614716"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "3.7"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "sqlalchemy": {
            "hashes": [
                "sha256:72325e67fb85f6e9ad304c603d83626d1df684fdf0c7ab1f0352e71feeab69d8"
            ],
            "index": "pypi",
            "version": "==1.2.10"
        }
    },
    "develop": {}
}

```
</details>

**Comment by techalchemy** [comment]
Nice. Thanks for the report!

**Comment by rakyi** [comment]
Unable to reproduce this on 2018.10.9.

**Comment by techalchemy** [comment]
This is fixed in 2018.10.13 for sure as I implemented canonicalizarion on both sides of the comparison of the comparison during cleaning — I think this actually tipped me off about it but I forgot to tag the issue when I fixed it. Thanks for the report!
